core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -69,6 +69,20 @@ object TaskContext {
   }
 }
 
+/**
+ * Identifies where an update for a data property accumulator update came from.  This is important
+ * to ensure that updates are not double-counted when rdds get recomputed.  When the executors send
+ * back data property acucmualtor updates, they seperate out the updates per rdd or shuffle output
+ * generated by the task. That gives the driver sufficient info to ensure that each update is
+ * counted once. The shuffleId is important since two seperate shuffle actions could happen on the
+ * same RDD and the RDD ID and partition ID with different accumulators.
+ * Any accumulators used inside of `runJob` directly are always counted because there is no
+ * resubmition of `runJob`/`foreach`.
```

Comment:
Can you add something here about why this is important for tracking incomplete partitions, eg. with repeatedly partially computing partitions with `take()`?